README.md

```diff
@@ -52,7 +52,7 @@ Example RESTful interaction:
 | Receive test result data                                       |  GET ``results``                                          |
 | Receive test KPIs                                              |  GET ``kpi``                                              |
 | Receive test case name                                         |  GET ``name``                                             |
-
+| Receive forecast						 						 |  GET ``forecasts``					     				 |
```

Comment:
@dhblum I've added the API call in here: https://github.com/ibpsa/project1-boptest/pull/62/commits/6fab1200752b6a0fb7a866a55cb20f5dfbf2e4e7 
However, this call would require arguments which I think is something that we have not had so far. Do you know how to include arguments in a an API call?

Reply:
Addressed in https://github.com/ibpsa/project1-boptest/pull/62/commits/6fab1200752b6a0fb7a866a55cb20f5dfbf2e4e7. Still need to add the arguments to though.

Reply:
Use a PUT request instead of GET.  Take a look at the implementation of the PUT request for the class ``Step`` in ``restapi.py``, which takes an argument.  Look at the instantiation and use of the ``parser_step`` object, which can accept multiple arguments if you define them.

Reply:
Hey @dhblum I just one concern: the way you do it with the control step I see that there is the need to assign to the `case` a variable called `step`. This is because the `put` and `get` are to different calls, not like a regular method where you just provide an argument and it will return you the outputs under the same call. I guess there is no way to avoid that so it is needed to add as many variables as arguments you want to provide. My questions are:

1. Do you want those variables (i.e. horizon, interval, category and plot) to belong to the `case` object or to the `forecaster` object.
2. Do you still want `plot` to be a possible argument? Not sure if it makes sense to have a `plot` state variable in the object. 
3. Should the default values come from the `config` file or from the `init`?

Reply:
@dhblum or it can be maybe approached with `POST` as it is done for `advance`? I see that `POST` does not require any attribute to be created for the arguments.

Reply:
Thanks @JavierArroyoBastida, good questions.  Here are my suggestions:

1.  The variables should belong to the ``case``.  
2.  I would not have ``plot`` be a possible argument.
3.  Default values should come from the ``config``.

Basically, I think it makes sense to treat forecast configuration parameters (horizon, interval, category) similar to how step is treated, except instead of a single float value, we are talking about a collection of values.  To accomplish, what I suggest is:

- [ ] Add a method ``TestCase.set_forecast_parameters()`` to store configuration parameters in ``TestCase`` (i.e. ``self``)
- [ ] Add a method ``TestCase.get_forecast_parameters()`` to retrieve the configuration parameters currently in ``TestCase`` (i.e. ``self``)
- [ ] Edit the method ``TestCase.get_forecast()`` to use the ``self.<parameter>`` in TestCase to retrieve the forecast.
- [ ] Edit the method ``TestCase.__init__()`` to use ``TestCase.set_forecast_parameters()`` with the ``config``.
- [ ] Add a class ``Forecast_Parameters`` to ``restapi.py`` with a ``GET`` method to retrieve current forecast parameters and ``PUT`` method to set new forecast parameters.  Use ``TestCase.set_forecast_parameters()`` and ``TestCase.get_forecast_parameters()`` as appropriate  Also add the associated URL extension ``'forecast_parameters'``
- [ ] Keep the class ``Forecast`` in ``restapi.py`` with the ``GET`` method as is.

Let me know if you think this makes sense or have any questions.

Reply:
@JavierArroyoBastida Neither ``PUT`` nor ``POST`` require attributes to be created in ``case``.  I've learned that typically, ``PUT`` implies you are updating already-present data, while ``POST`` implies you are creating new data.  The reason ``step`` is an attribute is because it is something I thought would only have to be set once, and would then just be used internally by ``case`` to move the simulation forward.  Then, the controller/user would not have to specify ``step`` each time they advance the simulation.  

What I suggest above is matching that style.  Otherwise, if you'd rather have the user specify the forecast parameters with each call, you can use a ``POST`` like ``advance``, with the appropriate arguments added.